src/backend/access/aocs/aocsam_handler.c

```diff
@@ -341,48 +352,41 @@ get_insert_descriptor(const Relation relation)
 
 	if (state->insertDesc == NULL)
 	{
-		List	*segments = NIL;
 		MemoryContext oldcxt;
 
+		/*
+		 * CBDB_PARALLEL:
+		 * Should not enable insertMultiFiles if the table is created by own transaction
+		 * or in utility mode.
+		 */
+		if (Gp_role != GP_ROLE_UTILITY &&
+			gp_appendonly_insert_files > 1 &&
+			!ShouldUseReservedSegno(relation, CHOOSE_MODE_WRITE))
+			state->insertMultiFiles = gp_appendonly_insert_files;
+
 		oldcxt = MemoryContextSwitchTo(aocoLocal.stateCxt);
 		state->insertDesc = aocs_insert_init(relation,
 									  ChooseSegnoForWrite(relation));
-		
-		dlist_init(&state->head);
-		dlist_head *head = &state->head;
-		dlist_push_tail(head, &state->insertDesc->node);
-
-		if (state->insertDesc->insertMultiFiles)
-		{
-			segments = lappend_int(segments, state->insertDesc->cur_segno);
-			for (int i = 0; i < gp_appendonly_insert_files - 1; i++)
-			{
-				next = aocs_insert_init(relation,
-												 ChooseSegnoForWriteMultiFile(relation, segments));
-				dlist_push_tail(head, &next->node);
-				segments = lappend_int(segments, next->cur_segno);
-			}
-			list_free(segments);
-		}
 
-		//* mark all insertDesc  placeholderInserted with false */
-        if (relationHasUniqueIndex(relation))
-        {
-            dlist_iter          iter;
-            dlist_foreach(iter, head)
-            {
-                AOCSInsertDesc insertDesc = (AOCSInsertDesc)dlist_container(AOCSInsertDescData, node, iter.cur);
-                insertDesc->placeholderInserted = false;
-            }
-        }
```

Comment:
I remove this codes try to set `insertDesc->placeholderInserted = false`,  beccuase aocs_insert_init() palloc0, there is no need to fetch everyone and set it to false again.

@lss602726449  could you confirm that?

Reply:
Yes, it's my mistake. Thanks for correcting.